### DISCUSSION TITLE
Possible contribution

### DISCUSSION THREAD

**Discussion by gbtorrance** [eprbell/rp2#90] [General]
I am interested in possibly contributing to `rp2` or `dali-rp2`. I am a senior software dev, but I am a beginner when it comes to Python. I've done some tutorials, and want to grow my Python skills, but I'm a bit "green". Is there a place for me to contribute? And if so, would `rp2` or `dali-rp2` be more appropriate? Thanks.

**Comment by macanudo527** [comment]
Thanks for jumping in! We appreciate any kind of help, and hopefully we can help you with your Python skills.

I would appreciate the help adding more exchanges to `dali-rp2` if you are interested and are using one of the exchanges we haven't implemented yet. 

If you are outside of the US or Japan, `rp2` needs reports for countries. There has been some interest in reports for Canada #52 , but no one with coding experience to do it.

**Comment by eprbell** [comment]
I second what @macanudo527 said. Thanks for offering to help and welcome to the project!

**Comment by gbtorrance** [comment]
Thanks for the replies, guys. I'll take some time to dig around and see what I might be able to take on. Just FYI, I likely won't have a lot of time, so progress may be slow, but hopefully I'll be able to contribute something of value over time. Thanks for the welcome.

**Comment by gbtorrance** [comment]
@macanudo527, @eprbell, I am based in the US. I have typically used Coinbase Pro, Binance US, and KuCoin, with CBP being my primary. (I've also used FTX, Gemini, and other CiFi platforms like Celsius and Voyager, all of which are now bankrupt. Good times.)

I have thought about possibly working on [this Coinbase Pro issue](https://github.com/eprbell/dali-rp2/issues/82) -- which is one that I originally submitted -- if you think it wouldn't be too complicated for a Python beginner.

**That said**, I'm curious about your thoughts on the future of Coinbase and Coinbase Pro in RP2. I understand CBP is going away soon and "advanced" features are going to be added to regular Coinbase (to eliminate the need for what CBP was offering). I assume "portfolios" (which is at the root of the above issue) will be supported in Coinbase, but I'm not certain.

Do you see value in continuing to work on anything CBP related, or is it better to just work on the Coinbase plugin instead?

I don't **need** to work on anything Coinbase related. I'm open to working on other things (including Binance US or KuCoin or whatever). Thoughts?

**Comment by eprbell** [comment]
Here's how I would sort the list in order of impact:
* implement Binance US, Gemini, KuCoin plugin (or other major exchange).
* Canadian plugin (#52): there are a bunch of people who expressed interest in this. Note that it's not necessary to be a Canadian tax expert: there are various people on that thread who can provide information on taxes over there. They have the tax knowledge but they are looking for a coder.
* implement Advanced Trading in Coinbase plugin (I was planning on doing some of that myself, but I'm happy to share the effort).
* CBPro portfolios: like you said CBPro is going away (I think I read they'll even disable the REST endpoint in a while), so this would be less impactful (but still OK to do if it's useful to you and you want to start there).

**Comment by gbtorrance** [comment]
Thanks of the insights. I think I'd like to try focus on one of the U.S.-based exchanges. (If I remember correctly, I had to use a VPN with KuCoin, and that was a pain so I'd rather avoid that.)

So I'm thinking one of the following (in no particular order of preference):
- BinanceUS
  - Would this be a variant of the Binance.com CCXT plugin that already exists?
- Gemini
  - I incorrectly assumed they had gone bankrupt, but I think I was confusing them with Genesis Lending, which has a relationship with Gemini via Gemini Earn? (And the Winklivoss twins are involved with both, I think?) Anyway...
  - Would this be a regular plugin or a CCXT one?
- Kraken
  - Regular or CCXT?

I'm OK with working on any of these. Do you have a preference? I'll go with whatever you prefer. Thanks.

**Reply by eprbell** [reply]
Kraken is being worked on by @ndopencode (see https://github.com/eprbell/dali-rp2/pull/114). Maybe go for Binance.us because it has a lot of users (but it's not a strong opinion: Gemini is good too). I recommend syncing up with @macanudo527 who wrote the Binance.com plugin (international version). All new plugins should be CCXT-based. Thanks!

**Comment by gbtorrance** [comment]
OK, sounds good @erpbell. Thanks. I'll start looking into Binance US and familiarizing myself with the Binance.com plugin. It will probably take me a while to before I'm ready to start coding anything, but I'll begin moving in that direction.

@macanudo527, if there is anything you think I should be aware of at this point, please let me know. And if there is a better forum for continuing this discussion, let me know that, too.

Thanks!

**Reply by eprbell** [reply]
No worries, take your time and if you have questions let us know. Maybe let's move the discussion to https://github.com/eprbell/dali-rp2/issues/4 (which also has all the links to get you started).

**Reply by gbtorrance** [reply]
Perfect! Thanks. Will do.

**Comment by haavardf** [comment]
Hi, I have just followed the setup on Linux, and run some unit tests. Since I havn't coded Phyton before there is a setup barrier for me. I assume you use an IDE, and I would like recomondations. Especially from someone that has time to answer questions, so maybe it would be good to use the same IDE.

**Reply by eprbell** [reply]
I suggest https://vscodium.com: it's very easy to use, very well-documented, open-source and doesn't have Microsoft telemetry/tracking (unlike VSCode).

**Reply by haavardf** [reply]
Installed, I will give it a try.

**Reply by macanudo527** [reply]
I'm a little late, but I use [Sublime](https://www.sublimetext.com/download). It's fairly modular and you can use Co-pilot with it if you have a subscription.
